### ISSUE TITLE
On iOS 14 MGLAnnotationView slows down map significantly

### ISSUE THREAD

**Issue by flashspys** [mapbox/mapbox-gl-native-ios#350]
The addition of even one custom MGLAnnotationView make the map unusable. The attached image shows the profiler where the mapbox rendering call is taking the most of the cpu. There is only a basic map showing one custom annotation. On iOS 13 this runs butter smooth. The Map is displayed via a UIViewRepresentable in SwiftUI.

![image](https://user-images.githubusercontent.com/1114404/88405358-700c4c80-cdcf-11ea-9058-de7462a168eb.png)

### Steps to reproduce

 1. On iOS 14, create a MGLMapView and display one custom MGLAnnotationView.

### Expected behavior
Map runs smooth

### Actual behavior
Map is unusable slow

### Configuration
The used custom MGLAnnotationView:

```swift
class CustomAnnotationView: MGLAnnotationView {
    override func layoutSubviews() {
        super.layoutSubviews()
        layer.cornerRadius = bounds.width / 2
        layer.borderWidth = 2
        layer.borderColor = UIColor.white.cgColor
    }
}
```
**Mapbox SDK version:** 6.0.0
**iOS versions:** iPad OS 14 / iOS 13.6
**Device/simulator models:** iPad Pro 1st Gen (iOS 14 device) / iPhone Xs (iOS 13 device)
**Xcode version:** 12.0 b2

**Comment by knov** [comment]
cc @tmpsantos @alexshalamov

**Comment by behdaad** [comment]
In my experience, even building with Xcode 11 (iOS SDK 13) and installing the IPA on a device running iOS 14 causes the same issue.

**Comment by mohammadZ74** [comment]
Same here, I have a bad lag on iOS 14 devices when showing Annotation. We haven't any problem on iOS 13 or earlier.

**Comment by julianrex** [comment]
@flashspys can you provide your sample app (looks like you're using our [mapbox-maps-swiftui-demo](https://github.com/mapbox/mapbox-maps-swiftui-demo))?

We have not been able to reproduce this yet, using the SwiftUI sample application, with many UIView annotations, and using your `CustomAnnotationView` above (built with Xcode 12 b5).

**Comment by flashspys** [comment]
@julianrex Thanks for your follow 😉
Please have a look at this commit: https://github.com/flashspys/mapbox-maps-swiftui-demo/commit/9c1963e09986a309a0ffc53b9b18ab0eacd95f48

This makes the map on my iPad Pro (10.5 inch) on iOS 14b5 unusable laggy. At the moment it got a bit better, but the panning and espacially the zooming is still horrible.

**Comment by julianrex** [comment]
@flashspys thanks - I just updated the device to beta 6, and now see the same issue.

**Comment by julianrex** [comment]
@flashspys can you confirm you first saw this with iOS 14 beta 2?

**Comment by julianrex** [comment]
This appears to be a continuation of an iOS [stuttering issue first seen in iOS 12.2](https://github.com/mapbox/mapbox-gl-native/issues/14232), related to synchronizing `UIView`s (most likely annotation views) to the map view rendering, enabled via `CAEAGLLayer.presentsWithTransaction` 😞 .

@flashspys @behdaad @mohammadZ74 (or anyone else suffering), while we continue to investigate solutions, can you please try setting:

`mapView.preferredFramesPerSecond = MGLMapViewPreferredFramesPerSecondLowPower`

and let us know if you continue to see any stuttering?

----
For reference:
- Similar thread on the developer forums, specifically talking about Metal: https://developer.apple.com/forums/thread/107302
- Open Radar for previous issue: http://www.openradar.me/radar?id=4926004078313472

**Comment by flashspys** [comment]
The issue continues with beta 6. Setting `mapView.preferredFramesPerSecond = MGLMapViewPreferredFramesPerSecond.lowPower` definitely helps, there is no stuttering like before, just the "normal" but constant 30-fps stuttering.

**Comment by arthurschiller** [comment]
Same issue here with iOS 14 Beta 6 and iPad Pro 2020. It seems a little better on my iPhone XS (same OS version), probably because of the lower framerate. Would be great to get this fixed before the iOS 14 release.

**Comment by swsong12** [comment]
Same issue here. IOS 14 Beta 6 (iphone XS Max)

**Comment by julianrex** [comment]
We have a PR (#411) that will land in the next release. We would appreciate it if you could test the release as soon as possible and let us know of any issues you see compared with `6.1.0`. Thanks!

**Comment by behdaad** [comment]
Thanks @julianrex, will definitely test it as soon as it's available.

**Comment by julianrex** [comment]
[6.2.0-beta.1](https://github.com/mapbox/mapbox-gl-native-ios/releases/tag/ios-v6.2.0-beta.1) has been released. You'll need to use Xcode 12 beta 6 if you intend to submit an app to TestFlight.

**Comment by swsong12** [comment]
Great 👍 It seems like this issue has been fixed (for us at least, iPhone XS MAX)

**Comment by jfcartkeep** [comment]
> We have a PR (#411) that will land in the next release. We would appreciate it if you could test the release as soon as possible and let us know of any issues you see compared with `6.1.0`. Thanks!

@julianrex Can this fix be done on the previous release `5.9.0`? If I use the stable release of iOS SDK for mapbox navigation, I am lock on this old release because of the `MapboxMobileEvents` requirement.
